Tools/RemoteSynchronization/Program.cs

```diff
@@ -0,0 +1,815 @@
+
+            // Unpack and parse the multi token options
+            Dictionary<string, string> global_options = (options["global-options"] as List<string>)
+                ?.Select(x => x.Split("="))
+                .ToDictionary(x => x[0], x => x[1])
+                ?? [];
+
+            // Parse the log level
+            var log_level = options["log-level"] as string ?? DEFAULT_LOG_LEVEL;
+            var log_level_parsed = Enum.TryParse<Duplicati.Library.Logging.LogMessageType>(log_level, true, out var log_level_enum);
+            log_level_enum = log_level_parsed ? log_level_enum : Duplicati.Library.Logging.LogMessageType.Information;
+
+            using var console_sink = new Duplicati.CommandLine.ConsoleOutput(Console.Out, global_options);
+            using var multi_sink = new Duplicati.Library.Main.ControllerMultiLogTarget(console_sink, log_level_enum, null);
+
+            // Parse the log file
+            var log_file = options["log-file"] as string ?? DEFAULT_LOG_FILE;
+            // The log file sink doesn't have to be disposed, as the multi_sink will take care of it
+            Duplicati.Library.Logging.StreamLogDestination? log_file_sink = null;
+            if (!string.IsNullOrEmpty(log_file))
+            {
+                string log_file_dir = SystemIO.IO_OS.PathGetDirectoryName(log_file);
+                if (!string.IsNullOrEmpty(log_file_dir) && !SystemIO.IO_OS.DirectoryExists(log_file_dir))
+                    SystemIO.IO_OS.DirectoryCreate(log_file_dir);
+                log_file_sink = new Duplicati.Library.Logging.StreamLogDestination(log_file);
+            }
+            multi_sink.AddTarget(log_file_sink, log_level_enum, null);
+
+            // Start the logging scope
+            using var _ = Duplicati.Library.Logging.Log.StartScope(multi_sink, log_level_enum);
+
+            Dictionary<string, string> src_opts = (options["src-options"] as List<string>)
+                ?.Select(x => x.Split("="))
+                .ToDictionary(x => x[0], x => x[1])
+                ?? [];
+            Dictionary<string, string> dst_opts = (options["dst-options"] as List<string>)
+                ?.Select(x => x.Split("="))
+                .ToDictionary(x => x[0], x => x[1])
+                ?? [];
+
+            // Merge the global options into the source and destination options
+            foreach (var x in global_options)
+                src_opts[x.Key] = dst_opts[x.Key] = x.Value;
+
+            // Check if we only had to parse the arguments
+            if (parse_only)
+            {
+                Duplicati.Library.Logging.Log.WriteInformationMessage(LOGTAG, "rsync", "Arguments parsed successfully; exiting.");
+                return 0;
+            }
+
+            // Load the backends
+            using var b1 = Duplicati.Library.DynamicLoader.BackendLoader.GetBackend(src, src_opts);
+            var b1s = b1 as IStreamingBackend;
+            System.Diagnostics.Debug.Assert(b1s != null);
```

Comment:
Consider throwing an exception or writing to console, so the user gets a nicer message.
In the Agent I am catching `UserInformationException` and printing the message, otherwise printing a stack trace as well.

Reply:
Right, this has been changed to log an error message with an exception, which is then thrown.